examples/booking_app/test/feature_hotel_search/view/hotel_search_page_test.dart

```diff
@@ -0,0 +1,151 @@
+import 'package:booking_app/base/common_blocs/hotel_manage_bloc.dart';
+import 'package:booking_app/base/common_blocs/hotels_extra_details_bloc.dart';
+import 'package:booking_app/feature_hotel_search/blocs/hotel_search_bloc.dart';
+import 'package:booking_app/feature_hotel_search/views/hotel_search_page.dart';
+import 'package:booking_app/lib_router/blocs/router_bloc.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:provider/provider.dart';
+
+import '../../feature_home/mock/hotel_manage_mock.dart';
+import '../../stubs.dart';
+import '../mock/hotel_search_mock.dart';
+import '../mock/hotels_extra_details_mock.dart';
+import '../mock/router_bloc_mock.dart';
+
+void main() {
+  group('HotelSearchPage', () {
+    final mockExtraDetailsBloc = hotelsExtraDetailsMockFactory();
+    final mockHotelManageBloc = hotelManageMockFactory();
+
+    testWidgets('Hotel sorting option opens/closes', (tester) async {
+      final mockRouterBloc = routerMockFactory();
+      final mockHotelSearchBloc = hotelSearchMockFactory(
+        hotels: Stub.paginatedListTwoHotels,
+      );
+      await tester.pumpWidget(MultiProvider(
+        providers: [
+          Provider<HotelSearchBlocType>.value(value: mockHotelSearchBloc),
+          Provider<RouterBlocType>.value(value: mockRouterBloc),
+          Provider<HotelsExtraDetailsBlocType>.value(
+              value: mockExtraDetailsBloc),
+        ],
+        child: const MaterialApp(
+          home: Scaffold(body: HotelSearchPage()),
+        ),
+      ));
+
+      // Tap on sort button to open sorting options
+      await tester.tap(find.byKey(const Key('sort_button')));
+      await tester.pump(const Duration(seconds: 1));
```

Comment:
https://api.flutter.dev/flutter/flutter_test/WidgetTester/pumpAndSettle.html

Reply:
When used in this context, a timeout happens. I've read somewhere online that in such case when closing an overlay, we can use the `pump` function which will advance the frame after some time (resulting in a closed overlay).